code/src/containers/tables/orders-table/components/orders-table-head/OrdersTableHead.tsx

```diff
@@ -1,15 +1,76 @@
+import { useState } from "react";
+import { useSearchParams } from "react-router-dom";
+
 import { AppTableCell } from "@/components/app-table/components";
+import AppTableSortLabel from "@/components/app-table/components/app-table-sort-label/AppTableSortLabel";
 import AppTypography from "@/components/app-typography/AppTypography";
 
 import "@/containers/tables/orders-table/components/orders-table-head/OrdersTableHead.scss";
 
 type OrderTableHeadProps = {
   head: string;
 };
+
 const OrdersTableHead = ({ head }: OrderTableHeadProps) => {
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const [sortDirections, setSortDirections] = useState<{
+    [key: string]: "asc" | "desc";
+  }>({
+    createdAt: "desc",
+    total: "desc",
+    orderStatus: "desc",
+    isPaid: "desc"
+  });
+
+  const getSortKey = () => {
+    switch (head) {
+      case "ordersTable.columns.createdAt":
+        return "createdAt";
+      case "ordersTable.columns.totalPrice":
+        return "total";
+      case "ordersTable.columns.status":
+        return "orderStatus";
+      case "ordersTable.columns.isPaid":
+        return "isPaid";
+      default:
+        return null;
+    }
+  };
```

Comment:
You can move it before this component to avoid rerenders or move it to separate file

Reply:
Moved above